packages/component-library/src/components/form/mt-switch/mt-switch.vue

```diff
@@ -1,394 +1,206 @@
 <template>
-  <div class="mt-field--switch__container">
-    <div class="mt-field--switch" :class="MtSwitchFieldClasses">
-      <div class="mt-field--switch__content">
-        <div class="mt-field--switch__input">
-          <!-- eslint-disable-next-line vuejs-accessibility/form-control-has-label -->
-          <input
-            :id="identification"
-            type="checkbox"
-            :name="formFieldName || identification"
-            :checked="inputState"
-            :disabled="isDisabled"
-            :aria-checked="inputState"
-            :aria-label="label"
-            role="checkbox"
-            @change.stop="onChange"
-          />
-          <div class="mt-field__switch-state">
-            <div class="mt-field__switch-state-knob" />
-          </div>
-        </div>
+  <div
+    :class="[
+      'mt-switch',
+      {
+        'mt-switch--no-top-margin': removeTopMargin,
+        'mt-switch--future-no-default-margin': !!futureFlags.removeDefaultMargin,
+        'mt-switch--not-bordered': !bordered,
+      },
+    ]"
+  >
+    <div
+      :class="[
+        'mt-switch__block',
+        {
+          'mt-switch__block--bordered': bordered,
+          'mt-switch__block--errored': !!error && !!bordered,
+        },
+      ]"
+    >
+      <input
+        :checked="isInherited ? inheritedValue : checked || modelValue"
+        type="checkbox"
+        :class="['mt-switch__switch', { 'mt-switch__switch--errored': !!error }]"
+        :id="id"
+        :disabled="disabled || isInherited"
+        :required="required"
+        :aria-describedby="!!error ? errorId : undefined"
```

Comment:
Can you add also a `aria-label` here? Otherwise many selectors in tests from my side would break and a hard to replace because I need to find the input somewhere close to the correct label